src/server.c

```diff
@@ -3731,6 +3737,13 @@ void call(client *c, int flags) {
             server.lua_caller->flags |= CLIENT_FORCE_AOF;
     }
 
+    /* Some commands may contain sensitive data that should
+     * not be available in the slowlog. */
+    if ((c->flags & CLIENT_PREVENT_LOGGING) && !(c->flags & CLIENT_BLOCKED)) {
```

Comment:
I think it looks a bit odd that we check the BLOCKED flag here, but if we don't then we must not clear the PREVENT flag here. 
I think it it be less fragile to clear the PREVENT flag right after logging, so we can then remove the BLOCKED flag here.

Reply:
an easy to fix it I think, we can use the `client_old_flags`

Reply:
I like soloestoy's suggestion

Reply:
I'm not sure either of these really make it clearer or less fragile. If the prevent flag is removed right after logging, we're relying on CMD_CALL_SLOWLOG and to always be set so that we go through that code path.

Reply:
We need to re factor the code doing the slowlog into one function that will be called from both place. Otherwise we have a risk of missing one of them when we update the other. 
I was planning to do that in https://github.com/redis/redis/pull/8632  (after this PR is merged). 
We can maybe do that refactoring here, but the Other PR also moves resetClient around, so maybe we should leave this cleanup to be reconsidered there..

Reply:
Since you already plan on doing refactoring, do you want to just do it in your PR then?

Reply:
sorry for the delay, yes. let's merge this as is, and consider further cleanup in the other PR.